producer-c/producer-cloudwatch-integ/canary/CanaryLogsUtils.cpp

```diff
@@ -26,10 +26,13 @@ STATUS initializeCloudwatchLogger(PCloudwatchLogsObject pCloudwatchLogsObject)
 
 VOID setUpLogEventVector(PCHAR logString)
 {
-    Aws::String awsCwString((Aws::String) logString);
-    auto logEvent =
-        Aws::CloudWatchLogs::Model::InputLogEvent().WithMessage(awsCwString).WithTimestamp(GETTIME() / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
-    gCloudwatchLogsObject->canaryInputLogEventVec.push_back(logEvent);
+    if(gCloudwatchLogsObject != NULL) {
```

Comment:
You should grab the lock prior to the check

Reply:
The lock belong to the object. If the object is NULL, this would segfault
Hence, the lock is acquired inside.

Reply:
The issue is that if this is accessed by two parallel threads, both could proceed with checking the gCloudwatchLogsObject and for both of the threads the object can be NULL so you will still have a condition of double init. 
If this function is the only place the initialization of the object happens and two threads can access it, you need to sequentialize the access. The easiest is to define a scoped lock and interlock the check as well

Reply:
That makes sense. Let me do that.